storage/innobase/buf/buf0lru.cc

```diff
@@ -417,9 +417,19 @@ buf_block_t *buf_LRU_get_free_block(bool have_mutex)
 	/* If there is a block in the free list, take it */
 	if ((block = buf_LRU_get_free_only()) != nullptr) {
 got_block:
+		const ulint available = UT_LIST_GET_LEN(buf_pool.free);
+		const ulint scan_depth = srv_LRU_scan_depth;
+
 		if (!have_mutex) {
 			mysql_mutex_unlock(&buf_pool.mutex);
 		}
+
+		if (UNIV_UNLIKELY(available < scan_depth)) {
+			mysql_mutex_lock(&buf_pool.flush_list_mutex);
```

Comment:
At the moment, `buf_pool.page_cleaner_status` is a non-atomic `unsigned` variable that is protected by `buf_pool.flush_list_mutex`. It looks like we could change it to `Atomic_relaxed<unsigned>` without too much trouble. All `+=` and `-=` have to be replaced with `=` so that they will use relaxed loads and stores instead of using atomic instructions, such as the 80486 `lock xadd`.

Reply:
Changing the variable to `Atomic_relaxed<unsigned>` is not trivial. It may result in some overhead in code like this:
```c++
      buf_pool.page_cleaner_set_idle(false);
      buf_pool.n_flush_inc();
```
With a regular `unsigned` variable, the compiler is allowed to generate code that only does one load and one store for the above. For `Atomic_relaxed<unsigned>`, it would have to emit instructions that access the memory bus several times. I think that this is too risky to do without careful analysis of the generated code.

Reply:
Agree. I think we can evaluate more in future.